### ISSUE TITLE
Compatibility with the new control flow syntax Angular

### ISSUE THREAD

**Issue by Nicoss54** [ghaschel/vscode-angular-html#89]
Currently the new control flow syntax (@if, @else, @swittch...) is not recognized by this plugins. 

Syntax is white for the moment, it will be awesome to have coloration syntax for that

Thanks for your amazing work

**Comment by ghaschel** [comment]
On it.

**Comment by ghaschel** [comment]
If someone is able to provide me the most complex template scenario (that uses all possible control flow, parameters, pipes and stuff like that), it'd be a terrific jumpstart and a huge help.

Also, just an FYI: I am going to start working on this but I believe this won't be as easy as the previous updates were. So it might take a while

**Comment by urielzen** [comment]
So far this is the most complex scenario I have in my code

![image](https://github.com/ghaschel/vscode-angular-html/assets/7947179/90e5ef1b-9b69-466c-a108-beeb24e2ca17)

**Comment by ghaschel** [comment]
Nice. Can you provide this snipped as text, please?

If you happen to encounter more scenarios please include it here and I'll use in the development

**Comment by urielzen** [comment]
```html
@if (
  {
    isLoading: isLoading$ignal(),
    standards: isLoading$ignal() && standards$ignal().length === 0 ? standardsLoadingPlaceholders : standards$ignal(),
    noStandardsMessage: noStandardsMessage$ignal(),
    lastGetStandardsErrored: lastGetStandardsErrored$ignal(),
    isSuperAdministrator: user$ignal().isSuperAdministrator
  };
  as vm
) {
  @if (vm.standards.length > 0) {
    <p-table [value]="vm.standards">
      <ng-template pTemplate="header">
        <tr>
          @for (c of columns; track c) {
            <th class="text-center">{{ c.header }}</th>
          }
        </tr>
      </ng-template>
      <ng-template
        pTemplate="body"
        [primeNgTableRow]="vm.standards"
        let-standard
        let-rowIndex="rowIndex">
        @if (!vm.isLoading) {
          <tr>
            @for (c of columns; track c) {
              @if (c.name === 'order') {
                <td class="py-0 w-4rem text-center">
                  <div>
                    <p-button
                      [rounded]="true"
                      [text]="true"
                      [icon]="PrimeIcons.CHEVRON_UP"
                      (onClick)="orderStandard(Direction.Forward, standard.extId)"
                      [disabled]="rowIndex === 0" />
                  </div>
                </td>
              }
              @if (c.name === 'actions1') {
                <td class="p-0 w-2rem text-center"></td>
              }
            }
          </tr>
        }
        @if (vm.isLoading) {
          <tr>
            @for (c of columns; track c) {
              <td><p-skeleton /></td>
            }
          </tr>
        }
      </ng-template>
    </p-table>
  }
  @if (vm.standards.length === 0) {
    @if (!vm.lastGetStandardsErrored) {
      <p-message
        severity="info"
        [text]="vm.noStandardsMessage" />
    }
    @if (vm.lastGetStandardsErrored) {
      <p-message
        severity="error"
        text="Error Loading Standards" />
    }
  }
}

```

**Comment by urielzen** [comment]
RFC from Angular 

https://github.com/angular/angular/discussions/51241

**Comment by ghaschel** [comment]
Well. This was easier and quicker than I ever thought it would. 

Grammar is working fine and it didn't break anything. Here are some previews:

![code4](https://github.com/ghaschel/vscode-angular-html/assets/5878866/d7b01875-26e1-43e0-80a3-3e8806ba3e9c)
![code5](https://github.com/ghaschel/vscode-angular-html/assets/5878866/0c9e372a-45bc-44f4-ae0f-e21a2ff8ca3d)
![code](https://github.com/ghaschel/vscode-angular-html/assets/5878866/b78b9e96-2a97-4482-89ee-d4d31831cfc6)
![code2](https://github.com/ghaschel/vscode-angular-html/assets/5878866/1e0b778e-9e98-4f3c-95f6-83fd75320d47)
![code3](https://github.com/ghaschel/vscode-angular-html/assets/5878866/0b53a244-1727-4f96-a9f6-66719d8f2cdc)

I'll be publishing the next version today still.

**Comment by Nicoss54** [comment]
> Well. This was easier and quicker than I ever thought it would.
> 
> Grammar is working fine and it didn't break anything. Here are some previews:
> 
> ![code4](https://user-images.githubusercontent.com/5878866/284629979-d7b01875-26e1-43e0-80a3-3e8806ba3e9c.png) ![code5](https://user-images.githubusercontent.com/5878866/284629996-0c9e372a-45bc-44f4-ae0f-e21a2ff8ca3d.png) ![code](https://user-images.githubusercontent.com/5878866/284630001-b78b9e96-2a97-4482-89ee-d4d31831cfc6.png) ![code2](https://user-images.githubusercontent.com/5878866/284630003-1e0b778e-9e98-4f3c-95f6-83fd75320d47.png) ![code3](https://user-images.githubusercontent.com/5878866/284630008-0b53a244-1727-4f96-a9f6-66719d8f2cdc.png)
> 
> I'll be publishing the next version today still.

The syntax of @defer is not correct here 

now the syntax is @defer(on viewport) I don’t know if it has an impact on the coloration syntax

**Comment by ghaschel** [comment]
I got that syntax from the RFC link you sent. I didn't realize that syntax didn't go through.

Now that you mentioned it, no, it won't  affect the coloration as it is.

![image](https://github.com/ghaschel/vscode-angular-html/assets/5878866/f4f6bba8-b4e6-487b-a1c8-9c8d1cbd7fc4)

**Comment by ghaschel** [comment]
This was implemented and it is available to download in v2.8.0.

Please, in order for the new coloring to be applied to the control flow syntax, either run the `Set legacy color customizations` command or add the colors via settings UI manually.

If anyone finds anything weird going on with the syntax coloring, please, feel free to open any other ticket. 

❤

**Comment by digeomel** [comment]
Hi, no matter what I do I cannot get the control flow keywords to be highlighted differently than normal text. I even tried manually setting `"vscode-angular-html.controlFlowKeyword": "#abcdef"`.

![Image](https://github.com/user-attachments/assets/e38c5ce0-afe1-4e19-98c0-ed46a3d10238)

This is the Ayu Mirage theme, but I tried with other themes and most are the same. Notice that some code inside the `@for` and `@if` blocks is properly highlighted, and the blocks are identified as such by the editor, but the colour is just not right.

What am I missing?
